main.go

```diff
@@ -45,6 +45,7 @@ func main() {
 
 	r.HandleFunc("/profiles/{key}", setup(controllers.GetProfileByKey))
 	r.HandleFunc("/profiles/{key}", setup(controllers.DeleteProfile))
+	r.HandleFunc("/profiles/{key}/lastChallenges/{limit}", setup(controllers.GetLastChallengesForProfile))
```

Comment:
This should be `/profiles/{key}/challenges?order=newest&limit=n` or similar., because you are in principle querying `/profiles/{key}/challenges`, just with a specific order and a specific number.